src/fu-main.c

```diff
@@ -1105,6 +1106,13 @@ fu_main_daemon_method_call (GDBusConnection *connection, const gchar *sender,
 	if (g_strcmp0 (method_name, "GetHostSecurityAttrs") == 0) {
 		g_autoptr(FuSecurityAttrs) attrs = NULL;
 		g_debug ("Called %s()", method_name);
+		if (priv->is_vm) {
+			g_dbus_method_invocation_return_error_literal (invocation,
+								       FWUPD_ERROR,
+								       FWUPD_ERROR_NOT_SUPPORTED,
+								       "HSI unavailable for hypervisor");
```

Comment:
I think the same thing should probably apply for containers too.

Reply:
Oof, that's harder. Do you know how to detect that?

Reply:
https://stackoverflow.com/questions/20010199/how-to-determine-if-a-process-runs-inside-lxc-docker seems to have a few ways to do it

Reply:
`systemd-detect-virt` has also implemented such checks: https://github.com/systemd/systemd/blob/5b14956385ee1076b98c3ca730bb9c1c9e95fc22/src/basic/virt.c#L450

Reply:
I've added something that should catch 99% of people trying this.